test/infrastructure/inmemory/pkg/server/mux.go

```diff
@@ -612,7 +612,7 @@ func (m *WorkloadClustersMux) DeleteWorkloadClusterListener(wclName string) erro
 	}
 
 	if wcl.listener != nil {
-		if err := wcl.listener.Close(); err != nil {
+		if err := wcl.listener.Close(); !errors.Is(err, net.ErrClosed) {
```

Comment:
@sivchari @fabriziopandini We are now hitting l.616 if err is nil

Reply:
Oops, I'd follow up it.